src/main/java/org/webshar/hrms/controller/RolePermissionController.java

```diff
@@ -0,0 +1,123 @@
+package org.webshar.hrms.controller;
+
+import java.util.Collections;
+import java.util.List;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.webshar.hrms.model.db.RolePermission;
+import org.webshar.hrms.request.rolepermission.RolePermissionCreateRequest;
+import org.webshar.hrms.request.rolepermission.RolePermissionUpdateRequest;
+import org.webshar.hrms.response.BatchResponse;
+import org.webshar.hrms.response.Response;
+import org.webshar.hrms.service.RolePermissionService;
+import org.webshar.hrms.service.exception.EntityAlreadyExistsException;
+import org.webshar.hrms.service.exception.EntityNotFoundException;
+import org.webshar.hrms.service.exception.ServiceException;
+
+@RestController
+@RequestMapping("/")
+public class RolePermissionController
+{
+
+  @Autowired
+  RolePermissionService rolePermissionService;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RolePermissionController.class);
+
+  @GetMapping(value = "/services/api/web/hrms/role_permissions/{role_permission_id}")
```

Comment:
As per REST API practice, hyphen is preferred over underscore for resource path.
e.g. /services/api/web/hrms/role-permissions instead of /services/api/web/hrms/role_permissions.
Need to update this and rest of the paths.

Also, I think the below API is more suitable in this case.
GET /services/api/web/hrms/roles/{role_id}/permissions.
Give a thought about this and implement accordingly.

Reply:
End point updated.